File: baek2805/2805.js
// const fs = require('fs');
// const input = fs.readFileSync('/dev/stdin').toString().trim().split('');

const input = ['4', '7', '20', '15', '10', '17']

const n = parseInt(input[0]);
const need = parseInt(input[0]);

let ans = 0;
let left = 0;
let right = parstInt(Math.max(...input))
let middle = right/2

while (left<=right){
    middle = (left+right)/2;
    let tree = 0;
    for(i=0; i<n; i++){
        if(middle<)
    }
}

for(i=0; i<input.length; i++){
    ans = ans + parseInt(input[i])-height
}
console.log(ans)

File: baek2805/2805.py
import sys

def input():
  return sys.stdin.readline().rstrip()
  
a,b = map(int, input().split())
li = list(map(int, input().split()))

left, right, ans = 0,max(li),0

while left<=right:
  middle = (left+right)//2
  tree = 0
  for i in range(a):
    if middle<li[i]:
      tree += li[i] - middle
  if tree >= b:
    ans = middle
    left = middle+1
  elif tree<b:
    right = middle-1
print(ans)